mangaki/mangaki/management/commands/compare.py

```diff
@@ -1,121 +1,75 @@
-        self.X_test = test[:, 0:2]
-        self.y_test = list(map(lambda choice: rating_values[choice], test[:, 2]))  # test[:, 2] for Movielens
-        if PIG_ID:
-            self.X_test = []
-            self.y_test = []
-            for work_id in range(self.nb_works):
-                self.X_test.append((PIG_ID, work_id))
-                self.y_test.append(0 if work_id not in pig_ratings else pig_ratings[work_id])
-            self.X_test = np.asarray(self.X_test)
-            self.y_test = np.asarray(self.y_test)
+    def load_dataset(self, dataset_name):
+        dataset = Dataset()
+        if dataset_name == 'movies':
+            dataset.load_csv('ratings-ml.csv')
+        else:
+            dataset.load_csv('ratings.csv', convert=lambda choice: rating_values[choice])
+        self.anonymized = dataset.anonymized
 
-    def execute(self):
-        for algo in self.algos:
-            print(algo)
-            algo.set_parameters(self.nb_users, self.nb_works)
-            # algo.load('backup.pickle')
-            algo.fit(self.X, self.y)
-            # algo.save('backup.pickle')
-            self.y_pred = algo.predict(self.X_test)
-            rmse = mean_squared_error(self.y_test, self.y_pred) ** 0.5
-            print('RMSE', rmse)
-            self.results.setdefault(algo.get_shortname(), []).append(rmse)
-
-    def display_ranking(self):
-        error = Counter()
-        print(len(self.X_test))
-        for rank, i in enumerate(sorted(range(len(self.X_test)), key=lambda i: -self.y_pred[i]), start=1):
-            if True or rank <= 100 or rank >= 8338 or self.X_test[i][1] == 491:
-                _, work_id = self.X_test[i]
-                if self.y_test[i]:
-                    print('%d. %s %f (was: %f)' % (rank, self.works[work_id], self.y_pred[i], self.y_test[i]))
-                    error[(self.works[work_id], self.y_pred[i], self.y_test[i])] = mean_squared_error([self.y_pred[i]], [self.y_test[i]])
-                else:
-                    print('%d. %s %f' % (rank, self.works[work_id], self.y_pred[i]))
-        print('Most deviated')
-        for (title, pred, true), _ in error.most_common(50):
-            if true < 0:
-                print(title, pred, true)
-
-    def display_chart(self):
-        handles = []
-        for algo in self.algos:
-            shortname = algo.get_shortname()
-            curve, = plt.plot(self.results['x_axis'], self.results[shortname], label=str(algo), linewidth=1 if shortname == 'svd' else algo.NB_NEIGHBORS / 15, color='red' if shortname == 'svd' else 'blue')
-            handles.append(curve)
-        plt.legend(handles=handles)
-        plt.show()
+    def compare_models(self):
+        k_fold = ShuffleSplit(n_splits=NB_SPLIT)
+        rmse_values = defaultdict(lambda: [])
+        for i_train, i_test in k_fold.split(self.anonymized.X):
+            for algo in self.algos:
+                model = algo()
+                print(model.get_shortname())
```

Comment:
As it is mainly used in a management command, I'd argue that:

- either, we should use the `logging` module, just instantiate a variable in the module named `logger = logging.getLogger(__name__)` and use `logger`.
- either, we should use Django's facilities for logging.